Model/User.php

```diff
@@ -554,4 +555,28 @@ public function removeGroup(GroupInterface $group)
 
         return $this;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEqualTo(BaseUserInterface $user)
```

Comment:
I'm totally missing the point here. What this method suppose to do and where is used/useful?

Reply:
The FOS `UserInterface` extends from `EquatableInterface`
https://api.symfony.com/master/Symfony/Component/Security/Core/User/EquatableInterface.html
afaik this is required for user switching

Reply:
> afaik this is required for user switching and re-authentication

I'm not totally sure of this but I really don't know it properly. Just wondering why `AdvantageUserInterface` didn't extends that interface aswell so that's the reason of my question.